File: typescript/core/src/health.ts
import packageJson from "../package.json" with { type: "json" };

export const HEALTH_PATH = "/.well-known/foldset";

export function buildHealthResponse(platform: string, sdkVersion: string): string {
  return JSON.stringify({
    status: "ok",
    core_version: packageJson.version,
    sdk_version: sdkVersion,
    platform,
    timestamp: new Date().toISOString(),
  });
}

File: python/core/foldset/health.py
from __future__ import annotations

import json
from datetime import datetime, timezone

from .config import PACKAGE_VERSION

HEALTH_PATH = "/.well-known/foldset"


def build_health_response(platform: str, sdk_version: str) -> str:
    return json.dumps({
        "status": "ok",
        "core_version": PACKAGE_VERSION,
        "sdk_version": sdk_version,
        "platform": platform,
        "timestamp": datetime.now(timezone.utc).isoformat(),
    })
